src/frontend/js/utils/CreditCardHelper/index.tsx

```diff
@@ -0,0 +1,39 @@
+import { CreditCard } from 'types/Joanie';
+
+export enum CreditCardExpirationStatus {
+  EXPIRED = 'expired',
+  SOON = 'soon',
+  FINE = 'fine',
+}
+
+export class CreditCardHelper {
+  /**
+   * Tells if the credit card will expire soon.
+   * Soon is considered below 3 months.
+   */
```

Comment:
Let's use advantage of classes. What do you think to add a private instance field creditCard to this class ? 

```
// Before
CreditCardHelper.isExpiredSoon(creditCard)
// After
CreditCardHelper(creditCard).isExpiredSoon()
```

```suggestion
export class CreditCardHelper {
  #creditCard
  
  constructor(creditCard: CreditCard) {
    this.#creditCard = creditCard;
  }
  /**
   * Tells if the credit card will expire soon.
   * Soon is considered below 3 months.
   */
```

Reply:
IMO a helper should not be associated with a specific entity. But if we want to go this way ( attaching an entity ), we should instead make `CreditCard` a class itself which implements those methods to really fit to OOP principles. 

I already thought of that but I saw that as a big paradigm change across the codebase which we require creating instances when fetching from the API. 

```
export class CreditCard {
  private title: string;
  // ...
  
  // `data` is the raw json from the api.
  constructor(data: any) {
    this.title = data.title;
  }
  
  public isExpired() {
    return ...
  }
}
```

Let's talk about it during a meeting if you want ? :)